workspaces/cli-shared/src/diffs/interaction-diff-worker-rust.ts

```diff
@@ -121,6 +122,14 @@ export class InteractionDiffWorkerRust {
     workerProcessOutput.pipe(fork([inMemorySink, fsSink]));
     return {
       results,
+      onWriteComplete: new Promise((resolve, reject) => {
+        fsSink.on('close', () => {
+          resolve();
+        });
+        fsSink.on('error', () => {
+          reject();
+        });
+      }),
```

Comment:
Is there a better way to promisify a stream?

Reply:
we can merge and let @JaapRood weigh in later, for now if we can't reproduce the issue then it should be good

Reply:
If just bothered with the file system sink, you'd use [`stream.finished`](https://nodejs.org/dist/latest-v14.x/docs/api/stream.html#stream_stream_finished_stream_options_callback). That handles a bunch of edge cases and also propagates the error rather than dropping it. In this case, that's what I would say is most appropriate here.